client/app/scripts/components/terminal.js

```diff
@@ -29,24 +27,35 @@ function ab2str(buf) {
 }
 
 function terminalCellSize(wrapperNode, rows, cols) {
-  //
-  // wrapperNode should be an unsized block. E.g. An element that 'clings' to
-  // its inner terminal so we get a nice size reading of the inner terminal.
-  //
-  const width = wrapperNode.clientWidth;
   const height = wrapperNode.clientHeight;
-  const pixelPerRow = height / rows;
+
+  // Guess the width of the row.
+  let width = wrapperNode.clientWidth;
+  // Now try and measure the first row we find.
+  const firstRow = wrapperNode.querySelector('.terminal div');
+  if (!firstRow) {
+    log("ERROR: Couldn't find first row, resizing might not work very well.");
+  } else {
+    const rowDisplay = firstRow.style.display;
+    firstRow.style.display = 'inline';
+    width = firstRow.offsetWidth;
```

Comment:
More testing and it uses &nbsp; to fill out the rest of the line so it won't wrap! \o/.

So this measurement will be accurate regardless of init ncols. I think.